### ISSUE TITLE
[BUG] Unexpected error "undeclared name"

### ISSUE THREAD

**Issue by nicolaasuni** [GoASTScanner/gas#113]
The latest master version with the 13th Jan changes (https://github.com/GoASTScanner/gas/commit/cc52ef5b26b37bf253625795b7de6de4d2ef82bd) is unexpectedly reporting the following error on one of my projects:
```
[gas] 2017/01/14 11:03:59 Processing "/home/user/GO/src/project/src/authrequest.go"...
[gas] 2017/01/14 11:03:59 Failed to process: "/home/user/GO/src/project/src/authrequest.go"
[gas] 2017/01/14 11:03:59 Error during type checking: "/home/user/GO/src/project/src/authrequest.go:25:22: undeclared name: SamlAuthnRequest"
[gas] 2017/01/14 11:03:59 Halting execution.
```
All previous gas tool versions works fine without reporting any error.
The  "SamlAuthnRequest" is a defined type struct.

**Comment by gcmurphy** [comment]
Thanks for reporting. It looks like the entire content of the package isn't being considered by the type checker. The simplest reproducer I can come up with is this: 

```go
// file1.go
package module

type Something struct {
	Foo string
	Bar string
}
```

```go
// file2.go
package module

import "fmt"

func FooBar(foo, bar string) {
	something := &Something{Foo: foo, Bar: bar}
	fmt.Printf("%v\n", something)
}
```

```
./gas ~/samples/module/...
[gas] 2017/01/14 13:53:14 Processing "/Users/gm/samples/module/file1.go"...
[gas] 2017/01/14 13:53:14 Processing "/Users/gm/samples/module/file2.go"...
[gas] 2017/01/14 13:53:14 Failed to process: "/Users/gm/samples/module/file2.go"
[gas] 2017/01/14 13:53:14 Error during type checking: "/Users/gm/samples/module/file2.go:6:16: undeclared name: Something"
[gas] 2017/01/14 13:53:14 Halting execution.
```
 
I'll submit a temporary fix that removes this as a fatal error so the previous behaviour is restored but the way files are processed might need to be reworked to fix it properly.

**Comment by gcmurphy** [comment]
Might need to use something like https://godoc.org/golang.org/x/tools/go/loader to address these types of errors.

**Comment by gcmurphy** [comment]
I am working on a fix for this in the [experimental branch](https://github.com/GoASTScanner/gas/tree/experimental)

**Comment by tiziano88** [comment]
FWIW I am also seeing the same issue

**Comment by gcmurphy** [comment]
This has been addressed in #146

**Comment by andy-miracl** [comment]
Thanks for fixing this...
